src/config/config_setup.cc

```diff
@@ -184,7 +184,9 @@ bool ConfigPathSetup::checkAgentPath(std::string& optValue)
 {
     std::string tmp_path;
     if (fs::path(optValue).is_absolute()) {
-        if (!isRegularFile(optValue) && !fs::is_symlink(optValue)) {
+        std::error_code ec;
+        fs::directory_entry dirEnt(optValue, ec);
+        if (!isRegularFile(dirEnt, ec) && !dirEnt.is_symlink(ec)) {
```

Comment:
I would rather use fs::is_symlink(dirEnt, ec) here and elsewhere.

It's more generic and helps reduce changes when refactoring.

Reply:
The point here is the sequence of `stats`. directory_entry encapsulates this.